roles/kubespray-defaults/defaults/main/download.yml

```diff
@@ -447,7 +447,7 @@ downloads:
       - etcd
 
   cni:
-    enabled: true
+    enabled: "{{ kube_network_plugin != 'calico' }}"
```

Comment:
Hi @VannTen, Why do we don't need to install cni when calico is enabled? I think we still need this, in some user cases, the user needs to run macvlan/ipvlan cni and calico at the same time.

Reply:
the CNI and calico installations this should be independent, not mutually exclusive. The similarity is that they both share the hostPath /opt/cni/bin. calico only installs the binary `calico` and `calico-ipam` into `/opt/cni/bin`, so it expects `/opt/cni/bin` to be present on the host, and if it is not, it creates.

It seems that revert  https://github.com/kubernetes-sigs/kubespray/pull/10407 should solve the problem, but I'm not sure what the owner variable does.

Reply:
>  but it seems calico is saying otherwise : https://github.com/projectcalico/calico/issues/6004#issue-1223490378

calico installs these CNI plugins: flannel, bandwidth, host-local, portmap, etc. These plugins may be required to work with calico (optional), so calico installs them by default. However, other plugins such as macvlan, ipvlan, etc. (https://github.com/containernetworking/plugins/tree/main/plugins) are not installed by calico. so we do need to install the whole cni plugins.

Reply:
> In that case, https://github.com/kubernetes-sigs/kubespray/commit/81ee96a1731761044ee0ff211cb594e56e473891 should suffice, I think 🤔

what the `recurse` variable does?

Reply:
It applies the file attribute on everything in that dir tree. see https://github.com/kubernetes-sigs/kubespray/issues/11747#issuecomment-2499200300

Reply:
Ok ! Thanks for the explanation, I haven't looked closely at the CNI bits in K8s before :+1: 

In that case, I think we should instead keep our install (but still fixes those recurse/permissions stuff) and patches the calico manifests to not install the CNI.
It seems their cni plugins are slightly different according to https://github.com/projectcalico/calico/issues/6004#issuecomment-2520931939 (bumped deps, etc). I'm not sure if we should ship their versions or not :thinking: when using calico.

The simpler path would be not to, that's for sure.

Reply:
> It applies the file attribute on everything in that dir tree. see https://github.com/kubernetes-sigs/kubespray/issues/11747#issuecomment-2499200300

the all CNI binary attributes must be **executable**,  or we can have a task for changing the file attributes under the `/opt/cni/bin`

> I think we should instead keep our install (but still fixes those recurse/permissions stuff) 

agree

> and patches the calico manifests to not install the CNI.

It's difficult for us, Calico, to install not only these CNI plugins: flannel, bandwidth, host-local, port map, etc., but also their own CNI plugins(Calico and calico-ipam).  It looks like the key to solving the problem is to change the properties of these files.